future.go

```diff
@@ -36,6 +36,12 @@ type IndexFuture interface {
 type ApplyFuture interface {
 	IndexFuture
 
+	// WaitCommitted blocks until the log entry has been committed to quorum.
+	// It does not wait for FSM application.
+	// The error returned follows the same semantics as the Error method,
+	// except for errors that occur after the log entry has been committed.
+	WaitCommitted() error
```

Comment:
However, there's an implicit dependency on the `applyCh` buffer size, which is controlled via the and `BatchApplyCh` and `MaxAppendEntries` configuration parameters (defaults to 64, max 1024, if `BatchApplyCh` is set to true). Once the buffer is full, `Apply` blocks.

```
	// MaxAppendEntries controls the maximum number of append entries
	// to send at once. We want to strike a balance between efficiency
	// and avoiding waste if the follower is going to reject because of
	// an inconsistent log.
```

I'd like to control the `applyCh` buffer size explicitly (as 1024 is fairly conservative) via a separate option, e.g., `AppendBufferSize` which would default to `MaxAppendEntries` and could not be less than that. It could replace the `BatchApplyCh` option.